ops/model.py

```diff
@@ -2998,6 +3042,40 @@ def secret_remove(self, id: str, *, revision: Optional[int] = None):
             args.extend(['--revision', str(revision)])
         self._run_for_secret('secret-remove', *args)
 
+    def open_port(self, protocol: str, port: Optional[int] = None):
+        arg = f'{port}/{protocol}' if port is not None else protocol
+        self._run('open-port', arg)
+
+    def close_port(self, protocol: str, port: Optional[int] = None):
+        arg = f'{port}/{protocol}' if port is not None else protocol
+        self._run('close-port', arg)
+
+    def opened_ports(self) -> List[OpenedPort]:
```

Comment:
Seems like this would be frequently used to compare between desired and actual. Should the return type be `Set[OpenedPort]` for easier comparison?
If OpenedPort is not hashable then should it be a namedtuple instead?

Reply:
Not a bad idea, as order isn't important (or guaranteed) here. I'll make that change.

`OpenedPort` is a dataclass with `frozen=True`, so it's hashable.